app/scripts/metamask-controller.js

```diff
@@ -7792,17 +7740,6 @@ export default class MetamaskController extends EventEmitter {
     };
   }
 
-  updateAccountBalanceForTransactionNetwork(transactionMeta) {
-    const {
-      networkClientId,
-      txParams: { from },
-    } = transactionMeta;
-    this.accountTrackerController.updateAccountByAddress({
-      address: from,
-      networkClientId,
-    });
-  }
```

Comment:
Same as with `KeyringController:stateChange`. This is just used to refresh balances reacting to `TransactionController:unapprovedTransactionAdded` and `TransactionController:transactionConfirmed`.

That should be done at controller level.